docs/source/cpp/api/builder.rst

```diff
@@ -31,9 +31,6 @@ Concrete builder subclasses
 .. doxygenclass:: arrow::BooleanBuilder
    :members:
 
-.. doxygenclass:: arrow::PrimitiveBuilder
```

Comment:
Is this expected to be here?

Reply:
This was causing a Breath warning and failing the CI build. `PrimitiveBuilder` is an implementation detail anyway (probably it should be in a `details` or `internal` namespace)